cas-server-core-audit/src/main/java/org/jasig/cas/audit/spi/TicketOrCredentialPrincipalResolver.java

```diff
@@ -115,4 +119,15 @@ private String resolveArgument(final Object arg1) {
         LOGGER.debug("Unable to determine the audit argument. Returning [{}]", UNKNOWN_USER);
         return UNKNOWN_USER;
     }
+
+    /**
+     * Default implementation that simply returns principal#id.
+     */
+    public static class DefaultPrincipalIdProvider implements PrincipalIdProvider {
```

Comment:
Or better yet a package-private. Whoever wants to subclass it would put the custom impl into `org.jasig.cas.audit.spi` package. Best of both worlds